### PR TITLE
[manila] add scheduler_hints to the shares CreateOpts

### LINKED ISSUE #3204: [manila] add scheduler_hints field to manila shares
https://docs.openstack.org/api-ref/shared-file-system/#create-share

Scheduler hints allow to create shares with affinity and anti-affinity rules, e.g.:

```json
{
    "share": {
...
        "scheduler_hints": {
            "same_host": "d9c66489-cf02-4156-b0f2-527f3211b243,4ffee55f-ba98-42d2-a8ce-e7cecb169182",
            "different_host": "903685eb-f242-4105-903d-4bef2db94be4"
        }
    }
}
```

### PR DESCRIPTION
**Pull request by kayrus**
Fixes #3204

### CODE REVIEW

**Comment by kayrus** [comment]
@pierreprinetti @mdbooth @EmilienM kindly ping

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 1]
```diff
@@ -48,6 +64,9 @@ type CreateOpts struct {
 	ConsistencyGroupID string `json:"consistency_group_id,omitempty"`
 	// The availability zone of the share
 	AvailabilityZone string `json:"availability_zone,omitempty"`
+	// SchedulerHints are hints for the scheduler to select the share backend
+	// Only available in and beyond API version 2.65
+	SchedulerHints *SchedulerHints `json:"scheduler_hints,omitempty"`
```
Comment (mdbooth) [inline openstack/sharedfilesystems/v2/shares/requests.go:69]: https://docs.openstack.org/api-ref/shared-file-system/#create-share ✅

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 2]
```diff
@@ -7,6 +7,22 @@ import (
 	"github.com/gophercloud/gophercloud/v2/pagination"
 )
 
+// SchedulerHints contains options for providing scheduler hints when creating
+// a Share.
+type SchedulerHints struct {
```
Comment (mdbooth) [inline openstack/sharedfilesystems/v2/shares/requests.go:12]: https://docs.openstack.org/api-ref/shared-file-system/#create-share ✅

**Review by mdbooth — COMMENTED** [review]
Marshalling looks correct to me.

/lgtm

**Review by pierreprinetti — APPROVED** [review]
Thank you! (and thank you @mdbooth for the review)

**Pull request merged**